src/Utils/Arrays.php

```diff
@@ -236,4 +236,26 @@ public static function normalize(array $arr, $filling = NULL)
 		return $res;
 	}
 
+	/**
+	 * Pop element from the array by key and return its value.
+	 * @param  array
+	 * @param  string|int array key
+	 * @param  mixed
+	 * @return mixed
+	 * @throws Nette\InvalidArgumentException if item does not exist and default value is not provided
+	 */
+	public static function pop(array & $arr, $key, $default = NULL)
+	{
+		if (is_array($arr) && array_key_exists($key, $arr)) {
+			$val = $arr[$key];
+			unset($arr[$key]);
+		} else {
+			if (func_num_args() < 3) {
+				throw new Nette\InvalidArgumentException("Missing item '$key'.");
+			}
+			return $default;
+		}
+		return $val;
```

Comment:
Why not? I see it working same way. If first argument is not array, you get provided default value like in get function. It could be odd to trigger error instead. So I decided is_array actually is important.